qmf/src/plugins/messageservices/imap/imapservice.cpp

```diff
@@ -1852,37 +1844,251 @@ void ImapService::updateStatus(const QString &text)
     updateStatus(QMailServiceAction::Status::ErrNoError, text, _accountId);
 }
 
+void ImapService::stopPushEmail()
+{
+    qMailLog(Messaging) << "Stopping push email for account" << _accountId
+                        << QMailAccount(_accountId).name();
+    _idling = false;
+    setPersistentConnectionStatus(false);
+    destroyIdleSession();
 #ifdef USE_KEEPALIVE
-void ImapService::onUpdateLastSyncTime()
+    _backgroundActivity->stop();
+#endif
+    _restartPushEmailTimer->stop();
+    _initiatePushEmailTimer->stop();
+}
+
+
+void ImapService::createIdleSession()
 {
-    // start timer again if still in idle mode
-    if (_idling) {
-        _backgroundActivity->wait();
-    } else if (_backgroundActivity->isRunning()){
-        qMailLog(Messaging) << Q_FUNC_INFO << "Stopping keepalive";
-        _backgroundActivity->stop();
+    if (!_networkConfigManager) {
+        _networkConfigManager = new QNetworkConfigurationManager(this);
+        Q_ASSERT(_networkConfigManager);
+        connect(_networkConfigManager, SIGNAL(onlineStateChanged(bool)),
+                    SLOT(onOnlineStateChanged(bool)));
+        _sessionTimer = new QTimer(this);
```

Comment:
_sessionTimer will be allocated repeatedly with connection errors, and I can't see any deletion.